File: qt/gen_qfontvariableaxis.go
package qt

import (
	"unsafe"
)

type QFontVariableAxis struct {
	h          uintptr
	isSubclass bool
}

// NewQFontVariableAxis constructs a new QFontVariableAxis object.
func NewQFontVariableAxis() *QFontVariableAxis {

	ret := newQFontVariableAxis(QFontVariableAxis_new())
	ret.isSubclass = true
	return ret
}

// NewQFontVariableAxis2 constructs a new QFontVariableAxis object.
func NewQFontVariableAxis2(axis *QFontVariableAxis) *QFontVariableAxis {

	ret := newQFontVariableAxis(QFontVariableAxis_new2(axis.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QFontVariableAxis) Swap(other *QFontVariableAxis) {
	QFontVariableAxis_Swap(this.h, other.cPointer())
}

func (this *QFontVariableAxis) OperatorAssign(axis *QFontVariableAxis) {
	QFontVariableAxis_OperatorAssign(this.h, axis.cPointer())
}

func (this *QFontVariableAxis) Tag() *QFont__Tag {
	_goptr := newQFont__Tag(QFontVariableAxis_Tag(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QFontVariableAxis) SetTag(tag QFont__Tag) {
	QFontVariableAxis_SetTag(this.h, tag.cPointer())
}

func (this *QFontVariableAxis) Name() string {
	var _ms struct_miqt_string = QFontVariableAxis_Name(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QFontVariableAxis) SetName(name string) {
	name_ms := struct_miqt_string{}
	name_ms.data = CString(name)
	name_ms.len = size_t(len(name))
	defer free(unsafe.Pointer(name_ms.data))
	QFontVariableAxis_SetName(this.h, name_ms)
}

func (this *QFontVariableAxis) MinimumValue() float64 {
	return (float64)(QFontVariableAxis_MinimumValue(this.h))
}

func (this *QFontVariableAxis) SetMinimumValue(minimumValue float64) {
	QFontVariableAxis_SetMinimumValue(this.h, (double)(minimumValue))
}

func (this *QFontVariableAxis) MaximumValue() float64 {
	return (float64)(QFontVariableAxis_MaximumValue(this.h))
}

func (this *QFontVariableAxis) SetMaximumValue(maximumValue float64) {
	QFontVariableAxis_SetMaximumValue(this.h, (double)(maximumValue))
}

func (this *QFontVariableAxis) DefaultValue() float64 {
	return (float64)(QFontVariableAxis_DefaultValue(this.h))
}

func (this *QFontVariableAxis) SetDefaultValue(defaultValue float64) {
	QFontVariableAxis_SetDefaultValue(this.h, (double)(defaultValue))
}

File: qt/gen_qfontvariableaxis.cpp
// +build ignore

#define WORKAROUND_INNER_CLASS_DEFINITION_QFont__Tag
#include <QFontVariableAxis>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qfontvariableaxis.h>
#include "gen_qfontvariableaxis.h"

#ifndef _Bool
#define _Bool bool
#endif

QFontVariableAxis* QFontVariableAxis_new() {
	return new QFontVariableAxis();
}

QFontVariableAxis* QFontVariableAxis_new2(QFontVariableAxis* axis) {
	return new QFontVariableAxis(*axis);
}

void QFontVariableAxis_Swap(QFontVariableAxis* self, QFontVariableAxis* other) {
	self->swap(*other);
}

void QFontVariableAxis_OperatorAssign(QFontVariableAxis* self, QFontVariableAxis* axis) {
	self->operator=(*axis);
}

QFont__Tag* QFontVariableAxis_Tag(const QFontVariableAxis* self) {
	return new QFont::Tag(self->tag());
}

void QFontVariableAxis_SetTag(QFontVariableAxis* self, QFont__Tag* tag) {
	self->setTag(*tag);
}

struct miqt_string QFontVariableAxis_Name(const QFontVariableAxis* self) {
	QString _ret = self->name();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QFontVariableAxis_SetName(QFontVariableAxis* self, struct miqt_string name) {
	QString name_QString = QString::fromUtf8(name.data, name.len);
	self->setName(name_QString);
}

double QFontVariableAxis_MinimumValue(const QFontVariableAxis* self) {
	qreal _ret = self->minimumValue();
	return static_cast<double>(_ret);
}

void QFontVariableAxis_SetMinimumValue(QFontVariableAxis* self, double minimumValue) {
	self->setMinimumValue(static_cast<qreal>(minimumValue));
}

double QFontVariableAxis_MaximumValue(const QFontVariableAxis* self) {
	qreal _ret = self->maximumValue();
	return static_cast<double>(_ret);
}

void QFontVariableAxis_SetMaximumValue(QFontVariableAxis* self, double maximumValue) {
	self->setMaximumValue(static_cast<qreal>(maximumValue));
}

double QFontVariableAxis_DefaultValue(const QFontVariableAxis* self) {
	qreal _ret = self->defaultValue();
	return static_cast<double>(_ret);
}

void QFontVariableAxis_SetDefaultValue(QFontVariableAxis* self, double defaultValue) {
	self->setDefaultValue(static_cast<qreal>(defaultValue));
}

void QFontVariableAxis_Delete(QFontVariableAxis* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QFontVariableAxis*>( self );
	} else {
		delete self;
	}
}

